modules/space/rendering/renderableconstellation.h

```diff
@@ -0,0 +1,123 @@
+/*****************************************************************************************
+ *                                                                                       *
+ * OpenSpace                                                                             *
+ *                                                                                       *
+ * Copyright (c) 2014-2022                                                               *
+ *                                                                                       *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this  *
+ * software and associated documentation files (the "Software"), to deal in the Software *
+ * without restriction, including without limitation the rights to use, copy, modify,    *
+ * merge, publish, distribute, sublicense, and/or sell copies of the Software, and to    *
+ * permit persons to whom the Software is furnished to do so, subject to the following   *
+ * conditions:                                                                           *
+ *                                                                                       *
+ * The above copyright notice and this permission notice shall be included in all copies *
+ * or substantial portions of the Software.                                              *
+ *                                                                                       *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,   *
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A         *
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT    *
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF  *
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE  *
+ * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                                         *
+ ****************************************************************************************/
+
+#ifndef __OPENSPACE_MODULE_SPACE___RENDERABLECONSTELLATION___H__
+#define __OPENSPACE_MODULE_SPACE___RENDERABLECONSTELLATION___H__
+
+#include <openspace/rendering/renderable.h>
+
+#include <modules/space/speckloader.h>
+#include <openspace/properties/optionproperty.h>
+#include <openspace/properties/selectionproperty.h>
+#include <openspace/properties/vector/vec3property.h>
+#include <openspace/properties/vector/ivec2property.h>
+#include <openspace/util/distanceconversion.h>
+#include <ghoul/opengl/ghoul_gl.h>
+#include <map>
+#include <vector>
+
+namespace ghoul::fontrendering { class Font; }
+namespace ghoul::opengl { class ProgramObject; }
+
+namespace openspace {
+
+namespace documentation { struct Documentation; }
+
+class RenderableConstellation : public Renderable {
```

Comment:
Maybe nitpicky, but should be plural form: `RenderableConstellations`?

And since this is a pure abstract base class, maybe name something like "RenderableConstellationsBase"? Thinking that could make it clearer that this renderable can not be instantiated? 🤔 Just a thought, open for suggestions

And add a descriptive comment above the class saying that this class is intended to be used as a base for the other constellation renderables :) [For example like here](https://github.com/OpenSpace/OpenSpace/blob/01eabd1b0286b068baa7c1639629b9ee1aabbd6b/modules/base/rendering/renderablenodeline.h#L44)